libsnark/polynomial_commitments/tests/polynomial_commitment_test_utils.hpp

```diff
@@ -0,0 +1,40 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+namespace libsnark
+{
+
+template<typename ppT>
+std::vector<libff::Fr<ppT>> gen_polynomial(const size_t degree)
+{
+    using Field = libff::Fr<ppT>;
+
+    const size_t num_random_values = std::min(4ul, degree);
+
+    std::vector<Field> coefficients;
+    coefficients.reserve(degree);
+
+    // First num_random_values are random (ensuring there are no zeroes)
+    for (size_t i = 0; i < num_random_values; ++i) {
+        coefficients.push_back(Field::random_element());
+        if (coefficients[i] == Field::zero()) {
```

Comment:
Why not use `.back()` to get a ref to the last element?

Reply:
Sure.  No reason not do do that.  Changed.